### ISSUE TITLE
support delivery of sitemap.xml from onedrive / gdrive

### ISSUE THREAD

**Issue by tripodsan** [adobe/helix-content-proxy#315]
as a first stab to tackle the sitemap.xml delivery problem, it should be possible to deliver `*sitemap*.xml` directly from onedrive / googledrive.

**Comment by tripodsan** [comment]
@trieloff since the sitemap is larger (1.7mb) than the 700k allowed by runtime, we can easily compress it (260k).
but then the universal adapter fails to detect the `isBinary`. so I can send it using `application/octet-stream`.

maybe it would be better to handle this transparently in the adapter. also, AWS allows 6mb responses and can also transparently compress the response:

https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-receive-response-with-compressed-payload.html

So instead of adding the compression in content-proxy, it probably better to do this in the adapter. WDYT?

**Comment by trieloff** [comment]
Adding compression in the wrapper needs to be done carefully so we don't break ESI

Maybe use https://github.com/adobe/helix-deploy/issues/151 instead?

Or have the content proxy push the sitemap to the blob store and serve it from there?

**Comment by tripodsan** [comment]
> Adding compression in the wrapper needs to be done carefully so we don't break ESI

true. but if the response is > 700k and it is openwhisk, then we hopefully don't have this in an ESI include.

> Maybe use [adobe/helix-deploy#151](https://github.com/adobe/helix-deploy/issues/151) instead?

maybe...

> Or have the content proxy push the sitemap to the blob store and serve it from there?

the sitemap is very volatile, we don't want to create too many blobs. eventually, it will probably be served from somewhere else. but we don't know where.

---

for now, I'll try if respecting the `content-encoding` in the `isBinary()` check for openwhisk solves the problem.
otherwise we deliver it `application/octet-stream` until we have a better idea.
